src/System.IO.FileSystem.DriveInfo/src/System.IO.FileSystem.DriveInfo.csproj

```diff
@@ -19,6 +19,12 @@
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Linux_Release|AnyCPU' " />
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'OSX_Debug|AnyCPU' " />
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'OSX_Release|AnyCPU' " />
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'FreeBSD_Debug|AnyCPU' "/>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'FreeBSD_Release|AnyCPU' "/>
+  <PropertyGroup>
+    <TargetsBSD Condition="'$(TargetsOSX)' == 'true' or '$(TargetsFreeBSD)' == 'true'">true</TargetsBSD>
+  </PropertyGroup>
```

Comment:
I still think the problem here is mixing names of APIs and platforms. So far it seemed to me that the Interop folders are for platforms, not for APIs. Because if we want to name the implementations by the origin of the APIs, we need a whole new naming scheme (starting with calling all POSIX APIs POSIX).

Also if we go for APIs, we must not have TargetOSX, TargetLinux etc. Then you can only define TargetsPOSIX2008, TargetsBSD4.4, TargetsWin32, etc. For any platform you then have to define the set of APIs that are implemented there.

Right now it is a big mess, we mix APIs and platforms, both in naming and implementations. Like this it just makes absolutely no sense. We have to decide for one system (either platforms or APIs) and stick with that. Not mix everything together.